### ISSUE TITLE
Nonsense: corim = tagged extensible type choice of tags 

### ISSUE THREAD

**Issue by deeglaze** [ietf-rats-wg/draft-ietf-rats-corim#333]
We need to remove the extensibility of $concise-rim-type-choice since extensibility is driven by profile, but profile isn't known at this point.

We also don't need to tag the type choice. Let's release the 500 tag?

`corim = concise-rim-type-choice`

**Comment by thomas-fossati** [comment]
> We need to remove the extensibility of $concise-rim-type-choice since extensibility is driven by profile, but profile isn't known at this point.

Indeed, that `$` looks like a typo to me.  `$concise-rim-type-choice` should be a closed type choice which, per [our typographical conventions](https://ietf-rats-wg.github.io/draft-ietf-rats-corim/draft-ietf-rats-corim.html#section-3), should read `concise-rim-type-choice`.

> We also don't need to tag the type choice. Let's release the 500 tag?
> 
> `corim = concise-rim-type-choice`

The two-level tagging system effectively smells like overengineering.

I don't recall why we originally decided to use this tagging scheme.

While I am not opposed to dropping it, this could produce wire-image incompatibility with the TCG's spec, which we agreed must be avoided.

**Comment by thomas-fossati** [comment]
> TCG Endorsement spec defines:
> 
> ```
> corim = #6.500($concise-reference-integrity-manifest-type-choice)
> tagged-corim-map = #6.501(corim-map)
> $concise-reference-integrity-manifest-type-choice /= tagged-corim-map
> $concise-reference-integrity-manifest-type-choice /= #6.502(signed-corim)
> 
> COSE-Sign1-corim = [
>   protected: bstr .cbor protected-corim-header-map
>   unprotected: unprotected-corim-header-map
>   payload: bstr .cbor tagged-corim-map
>   signature: bstr
> ]
> ```
> 
> The TCG is doing errata for this spec. Maybe it makes sense to influence this? But it would need to happen quickly and be minimally invasive.

Then I'd do:

```
corim = tagged-corim-map / tagged-cose-sign1-corim
```

which uses 501 for the unprotected and 18 for the sign1-protected `corim-map`.

**Comment by deeglaze** [comment]
Is it common for 

> TCG Endorsement spec defines:
> 
> ```
> corim = #6.500($concise-reference-integrity-manifest-type-choice)
> tagged-corim-map = #6.501(corim-map)
> $concise-reference-integrity-manifest-type-choice /= tagged-corim-map
> $concise-reference-integrity-manifest-type-choice /= #6.502(signed-corim)
> 
> COSE-Sign1-corim = [
>   protected: bstr .cbor protected-corim-header-map
>   unprotected: unprotected-corim-header-map
>   payload: bstr .cbor tagged-corim-map
>   signature: bstr
> ]
> ```
> 
> The TCG is doing errata for this spec. Maybe it makes sense to influence this? But it would need to happen quickly and be minimally invasive.

Which working group?

**Comment by thomas-fossati** [comment]
> > The TCG is doing errata for this spec. Maybe it makes sense to influence this? But it would need to happen quickly and be minimally invasive.
> 
> Which working group?

Should be DICE.

**Comment by nedmsmith** [comment]
> Then I'd do:
> 
> ```
> corim = tagged-corim-map / tagged-cose-sign1-corim
> ```

The TCG spec doesn't define `tagged-cose-sign1-corim` only `signed-corim = #6.18(COSE-Sign1-corim)`
But I think it is effectively the same thing. Essentially, the updated TCG spec would look like:
```
corim = concise-rim-type-choice
tagged-corim-map = #6.501(corim-map)
concise-rim-type-choice /= tagged-corim-map
concise-rim-type-choice /= #6.502(signed-corim)
```
It would be an extension to add:
```
concise-rim-type-choice /= signed-corim
```

**Comment by yogeshbdeshpande** [comment]
Removed 500 and 502 tags so closing the issue!
